src/examples/src/ActiveLink.ts

```diff
@@ -0,0 +1,72 @@
+import { create, diffProperty, invalidator, w } from '@dojo/framework/core/vdom';
+import { Handle } from '@dojo/framework/core/Destroyable';
+import injector from '@dojo/framework/core/middleware/injector';
+import cache from '@dojo/framework/core/middleware/cache';
+import { SupportedClassName } from '@dojo/framework/core/interfaces';
+import Link, { LinkProperties } from '@dojo/framework/routing/Link';
+import Router from '@dojo/framework/routing/Router';
+import { Params } from '@dojo/framework/routing/interfaces';
+
+export interface ActiveLinkProperties extends LinkProperties {
+	activeClasses: SupportedClassName[];
+	matchParams?: Params;
+}
+
+function paramsEqual(linkParams: any = {}, contextParams: any = {}) {
+	return Object.keys(linkParams).every((key) => linkParams[key] === contextParams[key]);
+}
+
+const factory = create({ injector, diffProperty, cache, invalidator }).properties<
+	ActiveLinkProperties
+>();
+
+export const ActiveLink = factory(function ActiveLink({
+	middleware: { diffProperty, injector, cache, invalidator },
+	properties,
+	children
+}) {
+	const { to, routerKey = 'router', params, matchParams = params } = properties();
+	let { activeClasses, classes = [], ...props } = properties();
+
+	diffProperty('to', (current: ActiveLinkProperties, next: ActiveLinkProperties) => {
+		if (current.to !== next.to) {
+			const router = injector.get<Router>(routerKey);
+			const currentHandle = cache.get<Handle>('handle');
+			if (currentHandle) {
+				currentHandle.destroy();
+			}
+			if (router) {
+				const handle = router.on('outlet', ({ outlet }) => {
```

Comment:
I'm guessing this validates that the path `to` points to is a valid path? Seems like this code block is here twice, maybe make a function for it?

Reply:
If I abstract it to a function, then I'll lose the type inference and have to import a type which I don't really want to do.